### PR TITLE
Create Smart widget XML

### PR DESCRIPTION
**Pull request by franciscodimattia**
I have no idea what I am doing, please review the widget and merge if acceptable.

### CODE REVIEW

**Comment by doktornotor** [comment]
I think you wanted to add the code from https://github.com/opnsense/plugins/pull/4214 to the PR here in the https://github.com/franciscodimattia/plugins/tree/patch-1 branch, not two do two separate PRs?

**Comment by franciscodimattia** [comment]
@doktornotor exactly, but doing it via the github website didn't allow me to create both patches in one PR

**Comment by doktornotor** [comment]
Just merge this: https://github.com/franciscodimattia/plugins/pull/1

File: sysutils/smart/src/opnsense/www/js/widgets/Smart.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export default class Smart extends BaseTableWidget {
```
Comment (fichtner) [inline sysutils/smart/src/opnsense/www/js/widgets/Smart.js:1]: I'll have to insist on a copyright header ;)

File: sysutils/smart/src/opnsense/www/js/widgets/Smart.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    $(`#${device}`).css({ color: health ? "green" : "red", fontSize: '150%' });
+                    $(`#${device}`).text(health ? "OK" : "FAILED");
+                } catch (error) {
+                    super.updateTable('smart-table', [[["Error"], $(`<span>${this.translations.nosmart} ${device}: ${error}</span>`).prop('outerHTML')]]);
+                }
+            }
+        }
+    }
+
+async onMarkupRendered() {
```
Comment (fichtner) [inline sysutils/smart/src/opnsense/www/js/widgets/Smart.js:58]:
```suggestion
    async onMarkupRendered() {
```

**Comment by fichtner** [comment]
Thanks, I'll merge and test next week.

**Comment by doktornotor** [comment]
image

Looks OK. (Cannot test with a failing drive. 😉 )

**Comment by doktornotor** [comment]
On another note @fichtner - are these widgets from plugins supposed to be covered by `Dashboard (all)` privileges? Does not seem to be the case ATM.

**Comment by fichtner** [comment]
@doktornotor yes the dashboard all is an unpractical leftover from the old times: we could add all the endpoints, but we would need to pull in the plugin ones as well which could be a maintenance nightmare:

https://github.com/opnsense/core/blob/d08069699f404e6ee6e6e2d702b769f832e504cb/src/opnsense/mvc/app/models/OPNsense/Core/ACL/ACL.xml#L54-L79

The idea is that the dashboard now feeds off the endpoint declaration themselves, so the dashboard shows and works when the user has permission for the parent page where the end points live. That seems to make the most sense.

So eventually all the old dashboard permissions need to go (with the individual ones already commented out).


Cheers,
Franco

**Comment by fichtner** [comment]
I've put this on hold for now, see c86e31a0 -- I'd rather not do the info call and the old widget was a bit simpler in that regard.

**Pull request merged**